hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionPruningLevel.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.execution.init;
+
+import java.util.EnumSet;
+
+/**
+ * The level of partition pruning that is required for a DAG to be executed.
+ * If member pruning is not possible, {@link PartitionPruningLevel#EMPTY_PRUNING} should be used.
+ * Levels are not excluding, and may be combined.
+ */
+public enum PartitionPruningLevel {
```

Comment:
we pass this as Enum which is returned to various clients (MC, java client, other clients) eg. via `Job.getConfig()`. Will it cause problems if they are unable to deserialize it? Should we use some other format (string, numeric id) to improve compatibility?

Reply:
after 2c93a4ffd49834e928a8ecba7799818d7b1d24aa will not be passed in JobConfig